addons/resource/models/resource_calendar_leaves.py

```diff
@@ -54,7 +54,7 @@ def _compute_company_id(self):
     @api.depends('date_from')
     def _compute_date_to(self):
         user_tz = timezone(self.env.user.tz or self._context.get('tz') or self.company_id.resource_calendar_id.tz or 'UTC')
-        for leave in self:
+        for leave in self.filtered(lambda l: l.date_from):
```

Comment:
I think it's best to avoid doing a first go-through with a filter, you can add a condition instead:
```suggestion
        for leave in self:
            if not leave.date_from:
                continue
```

Reply:
Done